airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/VersioningAcceptanceTests.java

```diff
@@ -0,0 +1,117 @@
+package io.airbyte.test.acceptance;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.invoker.generated.ApiClient;
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.airbyte.api.client.model.generated.DestinationDefinitionCreate;
+import io.airbyte.api.client.model.generated.DestinationDefinitionIdRequestBody;
+import io.airbyte.api.client.model.generated.DestinationDefinitionRead;
+import io.airbyte.api.client.model.generated.SourceDefinitionCreate;
+import io.airbyte.api.client.model.generated.SourceDefinitionIdRequestBody;
+import io.airbyte.api.client.model.generated.SourceDefinitionRead;
+import io.airbyte.test.utils.AirbyteAcceptanceTestHarness;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.SQLException;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class VersioningAcceptanceTests {
+
+  private static AirbyteApiClient apiClient;
+  private static UUID workspaceId;
+
+  private static AirbyteAcceptanceTestHarness testHarness;
+
+  @BeforeAll
+  static void init() throws ApiException, URISyntaxException, IOException, InterruptedException {
+    apiClient = new AirbyteApiClient(
+        new ApiClient().setScheme("http")
+            .setHost("localhost")
+            .setPort(8001)
+            .setBasePath("/api"));
+
+    workspaceId = apiClient.getWorkspaceApi().listWorkspaces().getWorkspaces().get(0).getWorkspaceId();
+
+    testHarness = new AirbyteAcceptanceTestHarness(apiClient, workspaceId);
+  }
+
+  @AfterAll
+  static void afterAll() {
+    testHarness.stopDbAndContainers();
+  }
+
+  @BeforeEach
+  void setup() throws SQLException, URISyntaxException, IOException {
+    testHarness.setup();
+  }
+
+  @AfterEach
+  void tearDown() {
+    testHarness.cleanup();
+  }
+
+
+  @ParameterizedTest
+  @CsvSource({
+      "2.1.1, 0.2.0",
+      "2.2.0, 0.2.1",
+  })
+  void testCreateSourceSpec(final String dockerImageTag, final String expectedProtocolVersion)
```

Comment:
It is not very clear about what is going to be feed in the parameter without looking a the doc. Am I the only one to feel that?

Reply:
Another option would be using a different source. I am not a fan of the extra link to follow to get to the test arguments.

```@ParameterizedTest
@MethodSource("stringIntAndListProvider")
void testWithMultiArgMethodSource(String str, int num, List<String> list) {
    assertEquals(5, str.length());
    assertTrue(num >=1 && num <=2);
    assertEquals(2, list.size());
}

static Stream<Arguments> stringIntAndListProvider() {
    return Stream.of(
        arguments("apple", 1, Arrays.asList("a", "b")),
        arguments("lemon", 2, Arrays.asList("x", "y"))
    );
}